collectors/proc.plugin/proc_stat.c

```diff
@@ -993,6 +993,13 @@ int do_proc_stat(int update_every, usec_t dt) {
         for(core = 0; core < schedstat_cores_found; core++) {
             if(unlikely(!(cpuidle_charts[core].active_time - cpuidle_charts[core].last_active_time))) {
                 pthread_t thread;
+                cpu_set_t global_cpu_set;
+
+                sched_getaffinity(getpid(), sizeof(cpu_set_t), &global_cpu_set);
```

Comment:
You should check the returned value of `sched_getaffinity` before testing the `global_cpu_set` variable.

Reply:
Also it would be better to use `pthread_getaffinity_np(pthread_self(), sizeof(cpu_set_t), &global_cpu_set)` to be consistent with the `pthread_setaffinity_np()` call.

Reply:
Ok I will change that!